pkg/schedulercache/resource_info.go

```diff
@@ -16,5 +16,71 @@ limitations under the License.
 
 package schedulercache
 
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/api/core/v1"
+)
+
 type Resource struct {
+	MilliCPU float64
+	Memory   float64
+}
+
+func EmptyResource() *Resource {
+	return &Resource{
+		MilliCPU: 0,
+		Memory:   0,
+	}
+}
+
+func (r *Resource) Clone() *Resource {
+	clone := &Resource{
+		MilliCPU: r.MilliCPU,
+		Memory:   r.Memory,
+	}
+	return clone
+}
+
+var minMilliCPU float64 = 10
+var minMemory float64 = 10 * 1024 * 1024
+
+func NewResource(rl v1.ResourceList) *Resource {
+	cpu := rl[v1.ResourceCPU]
+	mem := rl[v1.ResourceMemory]
+
+	return &Resource{
+		MilliCPU: float64(cpu.MilliValue()),
+		Memory:   float64(mem.Value()),
+	}
+}
+
+func (r *Resource) IsEmpty() bool {
+	return r.MilliCPU < minMilliCPU && r.Memory < minMemory
+}
+
+func (r *Resource) Add(rr *Resource) *Resource {
+	r.MilliCPU += rr.MilliCPU
+	r.Memory += rr.Memory
+	return r
+}
+
+func (r *Resource) Sub(rr *Resource) *Resource {
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
```

Comment:
suggest check the case `r.MilliCPU < rr.MilliCPU` and `r.Memory < rr.Memory`

Reply:
In the case when both (r.MilliCPU and r.Memory) are less than their counterparts, what should I do with the result? Would it be ok to print out that the resource is not available?

Reply:
sure, I think the print log is enough now

Reply:
In MVP, I'd suggest to panic it directly; and then, we need to know how to handle this exception.

For the panic, please create a wrapper func; so we can only log error info in future without go through the whole code.

Reply:
Got it :) Thanks for the suggestions!